bsp/esp32_s3_korvo_1/esp32_s3_korvo_1.c

```diff
@@ -0,0 +1,198 @@
+        .adc_button_config.button_index = BSP_BUTTON_VOLDOWN,
+        .adc_button_config.min = 720, // middle is 820mV
+        .adc_button_config.max = 920,
+    },
+    {
+        .type = BUTTON_TYPE_ADC,
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
+        .adc_button_config.adc_handle = &bsp_adc_handle,
+#endif
+        .adc_button_config.adc_channel = ADC_CHANNEL_7, // ADC1 channel 7 is GPIO8
+        .adc_button_config.button_index = BSP_BUTTON_VOLUP,
+        .adc_button_config.min = 280, // middle is 380mV
+        .adc_button_config.max = 480,
+    }
+};
+
+esp_err_t bsp_iot_button_create(button_handle_t btn_array[], int *btn_cnt, int btn_array_size)
+{
+    esp_err_t ret = ESP_OK;
+    if ((btn_array_size < BSP_BUTTON_NUM) ||
+            (btn_array == NULL)) {
+        return ESP_ERR_INVALID_ARG;
+    }
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
+    /* Initialize ADC and get ADC handle */
+    BSP_ERROR_CHECK_RETURN_NULL(bsp_adc_initialize());
+    bsp_adc_handle = bsp_adc_get_handle();
+#endif
+
+    if (btn_cnt) {
+        *btn_cnt = 0;
+    }
+    for (int i = 0; i < BSP_BUTTON_NUM; i++) {
+        btn_array[i] = iot_button_create(&bsp_button_config[i]);
+        if (btn_array[i] == NULL) {
+            ret = ESP_FAIL;
+            break;
+        }
+        if (btn_cnt) {
+            (*btn_cnt)++;
+        }
+    }
+    return ret;
+}
+
+static const led_strip_config_t bsp_leds_rgb_strip_config = {
+    .strip_gpio_num = BSP_LED_RGB_GPIO,   // The GPIO that connected to the LED strip's data line
+    .max_leds = BSP_LED_NUM,                  // The number of LEDs in the strip,
+    .led_pixel_format = LED_PIXEL_FORMAT_GRB, // Pixel format of your LED strip
+    .led_model = LED_MODEL_WS2812,            // LED strip model
+    .flags.invert_out = false,                // whether to invert the output signal
+};
+
+static const led_strip_rmt_config_t bsp_leds_rgb_rmt_config = {
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+    .rmt_channel = 0,
+#else
+    .clk_src = RMT_CLK_SRC_DEFAULT,        // different clock source can lead to different power consumption
+    .resolution_hz = 10 * 1000 * 1000,     // RMT counter clock frequency = 10MHz
+    .flags.with_dma = false,               // DMA feature is available on ESP target like ESP32-S3
+#endif
+};
+
+static led_indicator_strips_config_t bsp_leds_rgb_config = {
+    .is_active_level_high = 1,
+    .led_strip_cfg = bsp_leds_rgb_strip_config,
+    .led_strip_driver = LED_STRIP_RMT,
+    .led_strip_rmt_cfg = bsp_leds_rgb_rmt_config,
+};
+
+static const led_indicator_config_t bsp_leds_config = {
+    .mode = LED_STRIPS_MODE,
+    .led_indicator_strips_config = &bsp_leds_rgb_config,
+    .blink_lists = bsp_led_blink_defaults_lists,
+    .blink_list_num = BSP_LED_MAX,
+};
+
+esp_err_t bsp_led_indicator_create(led_indicator_handle_t led_array[], int *led_cnt, int led_array_size)
```

Comment:
This function is supposed to return an array of led_indicators right?

I can see that outpu parameter `led_cnt` is not set in the function, the input parameter `led_array_size` is not used, and only 1 led_indicator is initialized. Is ther esomething that I'm missing?

Reply:
In this case LED_STRIPS_MODE is used, with that you can control all the leds. This api is to conform with gpio mode config, where you could have more elements in the array.

Reply:
Fair enough, could you please write a note about this to the docs block of this function in .h file? Rest LGTM